snforge_std/src/errors.cairo

```diff
@@ -0,0 +1,31 @@
+use core::option::OptionTrait;
+use core::starknet::SyscallResult;
+use core::byte_array::BYTE_ARRAY_MAGIC;
+
+pub enum PanicDataOrString {
+    String: ByteArray,
+    PanicData: Array<felt252>,
+}
+
+type ExtendedSyscallResult<T> = Result<T, PanicDataOrString>;
+
+trait SyscallResultStringErrorTrait<T> {
+    fn map_string_errors(self: SyscallResult<T>) -> ExtendedSyscallResult<T>;
+}
+
+impl SyscallResultStringErrorTraitImpl<T> of SyscallResultStringErrorTrait<T> {
+    fn map_string_errors(self: SyscallResult<T>) -> ExtendedSyscallResult<T> {
+        match self {
+            Result::Ok(x) => Result::Ok(x),
+            Result::Err(panic_data) => {
+                if panic_data.len() > 0 && *panic_data.at(0) == BYTE_ARRAY_MAGIC {
```

Comment:
I guess we don't care about stupid use cases like `assert!(function(), "")` :D

Reply:
We care: an empty string would be serialized to `array![BYTE_ARRAY_MAGIC]`